src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1188,18 +1188,21 @@ public void OpenFileFromPath(string xmlPath, bool forceManualExecutionMode = fal
                         OnWorkspaceOpening(xmlDoc);
 
                         // TODO: #4258
-                        // Remove this ResetEngine call when multiple home workspaces is supported.
-                        // This call formerly lived in DynamoViewModel
-                        ResetEngine();
-
-                        // TODO: #4258
                         // The following logic to start periodic evaluation will need to be moved
                         // inside of the HomeWorkspaceModel's constructor.  It cannot be there today
                         // as it causes an immediate crash due to the above ResetEngine call.
```

Comment:
Sorry, don't get your point. What comment says is moving `StartPeriodicEvaluation()` to `HomeWorkspaceModel` constructor will cause crash. It is because home work space model is constructed before `ResetEngine()`, so it will have an invalid engine controller after `ResetEngine()`.